dowhy/causal_refuters/bootstrap_refuter.py

```diff
@@ -1,184 +1,208 @@
+
+                    if variable not in self._variables_of_interest:
+                        self.logger.error("The variable {} is not in {}".format(variable, self._variables_of_interest))
+                        raise ValueError("The variable selected by the User is not a confounder, Instrument Variable or a Effect Modifier")    
+        
+        elif required_variables is None:
+            self.logger.info("No required variable. Resorting to Default Behavior: Run bootstrapping without any change to the original data.")
+
+        else:
+            self.logger.warning("Incorrect type: {}. Expected an int or list".format( type(required_variables) ) )
+            required_variables = None
+
+        
+        self._chosen_variables = self.choose_variables(required_variables)
+
+        if self._chosen_variables is None:
+            self.logger.info("INFO: There are no chosen variables")
+        else:    
+            self.logger.info("INFO: The chosen variables are: " +
+                            ",".join(self._chosen_variables))
+
+        if self._probability_of_change is None:
+            if self._noise > 1:
+                self.logger.error("Error in using noise:{} for Binary Flip. The value is greater than 1".format(self._noise))
+                raise ValueError("The value for Binary Flip cannot be greater than 1")
+            else:
+                self._probability_of_change = self._noise
+        elif self._probability_of_change > 1:
+            self.logger.error("The probability of flip is: {}, However, this value cannot be greater than 1".format(self._probability_of_change))
+            raise ValueError("Probability of Flip cannot be greater than 1")
+
+    
+    def refute_estimate(self, *args, **kwargs):
+        if self._sample_size > len(self._data):
+                self.logger.warning("The sample size is larger than the population size")
+
+        sample_estimates = np.zeros(self._num_simulations)
+        self.logger.info("Refutation over {} simulated datasets of size {} each"
+                         .format(self._num_simulations
+                         ,self._sample_size )
+                        ) 
+        
+        for index in range(self._num_simulations):
+            if self._random_state is None:
+                new_data = resample(self._data, 
+                                n_samples=self._sample_size )
+            else:
+                new_data = resample(self._data,
+                                    n_samples=self._sample_size,
+                                    random_state=self._random_state )
+
+
+            if self._chosen_variables is not None:
+                for variable in self._chosen_variables:
+                    
+                    if ('float' or 'int') in new_data[variable].dtype.name:
+                        scaling_factor = new_data[variable].std() 
+                        new_data[variable] += np.random.randn(self._sample_size) * self._noise * scaling_factor
```

Comment:
It may be more interpretable to instead sample directly from a normal distribution with mean 0 and sigma = scaling_factor*self._noise. 

Multiplication of standard normal has a less clear interpretation, at least for me.